File: LongestSubStringWithoutRepeatingCharacters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int leftPointer = 0, rightPointer = 0, max = 0;
        HashSet<Character> set = new HashSet<>();
        while (rightPointer < s.length()){
            if (!set.contains(s.charAt(rightPointer))){
                set.add(s.charAt(rightPointer));
                rightPointer++;
            }else {
                max = Math.max(max, set.size());
                leftPointer ++;
                rightPointer = leftPointer;
                set.clear();
            }
        }
        max = Math.max(max, set.size());
        return max;
        
        
    }
}

File: LongestSubStringWithoutRepeatingCharacters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        rememberingList = set()
        maxSubStringLength = 0
        left = 0
        for right in s:
            while right in rememberingList:
                rememberingList.remove(s[left])
                left += 1
            rememberingList.add(right)
            maxSubStringLength = max(maxSubStringLength, len(rememberingList))
        return max(maxSubStringLength, len(rememberingList))
